armi/utils/customExceptions.py

```diff
@@ -0,0 +1,163 @@
+
+    def __init__(self, msg):
+        self.msg = msg
+        self.caller = getframeinfo(stack()[1][0])
+
+    def __str__(self):
+        # Check if the call site is sensible enough to warrant printing. For now making the wild assumption that cython
+        # will wrap the fake stack filename in <>
+        callSiteIsFake = self.caller.filename.startswith(
+            "<"
+        ) and self.caller.filename.endswith(">")
+        if callSiteIsFake:
+            return self.msg
+        else:
+            return (
+                self.caller.filename + ":" + str(self.caller.lineno) + " - " + self.msg
+            )
+
+
+# ---------------------------------------------------
+
+
+class SettingException(Exception):
+    """Standardize behavior of setting-family errors"""
+
+    def __init__(self, msg):
+        Exception.__init__(self, msg)
+
+
+class InvalidSettingsStopProcess(SettingException):
+    """Exception raised when setting file contains invalid settings and user aborts or process is uninteractive"""
+
+    def __init__(self, reader):
+        msg = "Input settings file {}".format(reader.inputPath)
+        if reader.liveVersion != reader.inputVersion:
+            msg += (
+                '\n\twas made with version "{0}" which differs from the current version "{1}." '
+                'Either create the input file with the "{1}", or switch to a development version of ARMI.'
+                "".format(reader.inputVersion, reader.liveVersion)
+            )
+        if reader.invalidSettings:
+            msg += (
+                "\n\tcontains the following {} invalid settings:\n\t\t"
+                "{}"
+                "".format(
+                    len(reader.invalidSettings), "\n\t\t".join(reader.invalidSettings)
+                )
+            )
+        SettingException.__init__(self, msg)
+
+
+class NonexistentSetting(SettingException):
+    """Exception raised when a non existent setting is asked for"""
+
+    def __init__(self, setting):
+        SettingException.__init__(
+            self, "Attempted to locate non-existent setting {}.".format(setting)
+        )
+
+
+class InvalidSettingsFileError(SettingException):
+    """Not a valid xml or settings file"""
+
+    def __init__(self, path, customMsgEnd=""):
+        msg = "Attempted to load an invalid settings file from: {}. ".format(path)
+        msg += customMsgEnd
+
+        SettingException.__init__(self, msg)
+
+
+class NonexistentSettingsFileError(SettingException):
+    """Settings file does not exist"""
+
+    def __init__(self, path):
+        SettingException.__init__(
+            self, "Attempted to load settings file, cannot locate file: {}".format(path)
+        )
+
+
+# ---------------------------------------------------
+
+
+@important
+def general_submittedJob(job):
+    return "Submitted {} ... {}".format(job, job.get_state())
+
+
+@important
+def general_waitingForJob(job):
+    return "Waiting for {} to complete.".format(job)
```

Comment:
Did these get pulled in from the defunct localization stuff? we should probably remove.

Reply:
I copied these over from the old localization module because they were being used in a bunch of plugins, yeah.